python/dgl/nn/pytorch/graph_transformer.py

```diff
@@ -0,0 +1,72 @@
+import torch
+import torch.nn as nn
+import dgl
+
+
+class DegreeEncoder(nn.Module):
+    r"""Degree Encoder, as introduced in
+    `Do Transformers Really Perform Bad for Graph Representation?
+    <https://proceedings.neurips.cc/paper/2021/file/f1c1592588411002af340cbaedd6fc33-Paper.pdf>`__
+    This module is a learnable degree embedding module.
+
+    Parameters
+    ----------
+    max_degree : int
+        Upper bound of degrees to be encoded. Each degree will be clamped to (0, max_degree).
+    embedding_dim : int
+        Dimension of output embedding vector of each node.
+    direction : str, optional
+        Degrees of which direction to be encoded, only in "in", "out" and "both".
+        Default : "both".
+
+    Example
+    ------
+    >>> import dgl
+    >>> from dgl.nn.pytorch import DegreeEncoder
+
+    >>> g = dgl.graph(([0,0,0,1,1,2,3,3], [1,2,3,0,3,0,0,1]))
+    >>> degree_encoder = DegreeEncoder(5, 16)
+    >>> degree_embedding = degree_encoder(g)
+    """
+    def __init__(self, max_degree, embedding_dim, direction='both'):
+        super(DegreeEncoder, self).__init__()
+        self.direction = direction
+        self.degree_encoder = nn.Embedding(max_degree + 1, embedding_dim, padding_idx=0)

```

Comment:
If the direction is `both`. You should initialize two embedding tables for `in` and `out`, respectively.